### PR TITLE
test doc: tests `acceptstalefeeestimates` option is only supported on regtest chain

### PR DESCRIPTION
**Pull request by ismaelsadeeq**
This PR Follow up comments from [#27622](https://github.com/bitcoin/bitcoin/pull/27622) 

It test that the new `regtest-only` option `acceptstalefeeestimates` is not supported on [main, signet and test chains](https://github.com/bitcoin/bitcoin/pull/27622/files#r1235218268), removes an unnecessary [comment](https://github.com/bitcoin/bitcoin/pull/27622/files#r1235204323), and update fee estimator  `MAXFILEAGE` [description comment](https://github.com/bitcoin/bitcoin/pull/27622/files#r1233887314).

### CODE REVIEW

**Comment by pinheadmz** [comment]
Due to the multiple datadirs in the `feature_config_args` test, this assertion actually *does* fail in `combine_logs.py`:

https://github.com/bitcoin/bitcoin/blob/db7120fbaddff6037734d8e42d1d002d773f8567/test/functional/combine_logs.py#L86

I get the same error on master branch too, so I dunno if it's really up to you and this PR to patch around it or not.

**Review by pinheadmz — COMMENTED** [review]
concept ACK db7120fbaddff6037734d8e42d1d002d773f8567

Good follow up, addressed the TODOs leftover from #27622 well. Left a few comments

File: test/functional/feature_config_args.py [hunk 1]
```diff
@@ -108,6 +108,29 @@ def test_config_file_parser(self):
         self.restart_node(0)
         self.nodes[0].stop_node(expected_stderr=f'Warning: {inc_conf_file_path}:1 Section [testnot] is not recognized.{os.linesep}{inc_conf_file2_path}:1 Section [testnet] is not recognized.')
 
+        with open(inc_conf_file_path, 'w', encoding='utf-8') as conf:
+            conf.write('')  # clear
```
[the remark this answers is not in the document]
Comment (ismaelsadeeq) [reply test/functional/feature_config_args.py]: Thank you.
I removed the clears.

**Comment by ismaelsadeeq** [comment]
> I get the same error on master branch too, so I dunno if it's really up to you and this PR to patch around it or not.

I think this should be fixed in follow-up PR.

File: test/functional/feature_config_args.py [hunk 2]
```diff
@@ -108,6 +108,23 @@ def test_config_file_parser(self):
...
+            conf.write('testnet=1\n')
+            conf.write('acceptstalefeeestimates=1\n')
+        self.nodes[0].assert_start_raises_init_error(extra_args=[f"-conf={inc_conf_file_path}", "-allowignoredconf"], expected_msg='Error: acceptstalefeeestimates is not supported on test chain.')
+
+        self.log.info("test acceptstalefeeestimates option is not supported on signet chain.")
+        with open(inc_conf_file_path, 'w', encoding='utf-8') as conf:
+            conf.write('signet=1\n')
+            conf.write('acceptstalefeeestimates=1\n')
+        self.nodes[0].assert_start_raises_init_error(extra_args=[f"-conf={inc_conf_file_path}", "-allowignoredconf"], expected_msg='Error: acceptstalefeeestimates is not supported on signet chain.')
+
```
[the remark this answers is not in the document]
Comment (ismaelsadeeq) [reply test/functional/feature_config_args.py]: Thank you, I updated with your suggestions

File: test/functional/feature_fee_estimation.py [hunk 3]
```diff
@@ -378,7 +378,7 @@ def test_acceptstalefeeestimates_option(self):
         os.utime(fee_dat, (last_modified_time, last_modified_time))
 
         # Restart node with -acceptstalefeeestimates option to ensure fee_estimate.dat file is read
-        self.start_node(0,extra_args=["-acceptstalefeeestimates"])
+        self.start_node(0, extra_args=["-acceptstalefeeestimates"])
```
[the remark this answers is not in the document]
Comment (ismaelsadeeq) [reply test/functional/feature_fee_estimation.py]: Done, thanks

File: src/policy/fees.cpp [hunk 4]
```diff
@@ -548,14 +548,14 @@ CBlockPolicyEstimator::CBlockPolicyEstimator(const fs::path& estimation_filepath
...
-    // Whenever the fee estimation file is not present return early
     if (est_file.IsNull()) {
         LogPrintf("%s is not found. Continue anyway.\n", fs::PathToString(m_estimation_filepath));
         return;
     }
 
     std::chrono::hours file_age = GetFeeEstimatorFileAge();
-    // fee estimate file must not be too old to avoid wrong fee estimates.
+    // Serving stale fee estimates can cause transactions to become stuck in the bottom of the mempool in the worst case.
+    // Don't do that.
```
Comment (glozow) [inline src/policy/fees.cpp:558]: I think I know what you mean, but imo transactions being stuck is a little bit of a jump from feerate underestimation. Maybe "Serving stale fee estimates may lead to overpaying in fees or not meeting minimum feerate requirements in the worst case." Or just delete this comment, as the code is self-explanatory and there is already a comment above `MAX_FILE_AGE`.

**Review by jonatack — COMMENTED** [review]
Concept ACK

File: test/functional/feature_config_args.py [hunk 5]
```diff
@@ -371,6 +371,18 @@ def test_ignored_default_conf(self):
...
         node.args = node_args
 
+    def test_acceptstalefeeestimates_arg_support(self):
+        self.log.info(f"Test -acceptstalefeeestimates option is not supported on non-regtest chains")
+        inc_conf_file_path = os.path.join(self.nodes[0].datadir, 'include.conf')
+        for chain in ["main", "test", "signet"]:
+            with open(inc_conf_file_path, 'w', encoding='utf-8') as conf:
+                conf.write(f'chain={chain}\n')
+                conf.write('acceptstalefeeestimates=1\n')
+            self.nodes[0].assert_start_raises_init_error(extra_args=[f"-conf={inc_conf_file_path}", "-allowignoredconf"], expected_msg=f'Error: acceptstalefeeestimates is not supported on {chain} chain.')
```
[the remark this answers is not in the document]
Comment (ismaelsadeeq) [reply test/functional/feature_config_args.py]: Thank you @jonatack, I was able to recreate the same Error.
b24ffb05e42ec13ade193c9d7bea0def4296a4a8 should fix this.

[the remark this answers is not in the document]
Comment (ismaelsadeeq) [reply test/functional/feature_config_args.py]: Thanks, Fix this  and the nit in  530ea157ea1c25821711617e5f1da3e95cfdc66d

**Comment by ismaelsadeeq** [comment]
rebased

**Review by jonatack — COMMENTED** [review]
Tested ACK 8884d5c6059495253dc84e1b984315ddbc78ecd3

Some non-blocking comments, happy to re-ack if you update.

File: test/functional/feature_config_args.py [hunk 6]
```diff
@@ -371,6 +371,18 @@ def test_ignored_default_conf(self):
             f'is being used instead.') + r"[\s\S]*", env=env, match=ErrorMatch.FULL_REGEX)
         node.args = node_args
 
+    def test_acceptstalefeeestimates_arg_support(self):
+        self.log.info("Test -acceptstalefeeestimates option support")
+
+        conf_file = self.nodes[0].datadir_path / "bitcoin.conf"
+        chains = {"main":"", "test":"testnet3", "signet":"signet"}
```
[the remark this answers is not in the document]
Comment (ismaelsadeeq) [reply test/functional/feature_config_args.py]: Thank you, I fixed the nit.

File: test/functional/feature_config_args.py [hunk 7]
```diff
@@ -371,6 +371,18 @@ def test_ignored_default_conf(self):
...
+        for chain, chain_name in chains.items():
+            util.write_config(conf_file, n=0, chain=chain_name, extra_config='acceptstalefeeestimates=1\n')
+            self.nodes[0].assert_start_raises_init_error(expected_msg=f'Error: acceptstalefeeestimates is not supported on {chain} chain.')
+
+        util.write_config(conf_file, n=0, chain="regtest")  # Reset to regtest
+
+
```
[the remark this answers is not in the document]
Comment (ismaelsadeeq) [reply test/functional/feature_config_args.py]: I removed the blank lines.

**Review by glozow — COMMENTED** [review]
utACK ee5a0369cc4305da7b3d26f37677de05ad797e51

**Pull request merged**